### PR TITLE
feat: support and use postcss@8

### PR DESCRIPTION
**Pull request by AviVahl**
- upgrade postcss and related deps.
- adapted type imports to the new v8 declarations.
- fixed all new lint errors (postcss is better typed now, so `.nodes` is defined correctly in more places, ridding the need for `!`)
- reported 3 parser regressions upstream (when using with `postcss-safe-parser`).

### CODE REVIEW

**Comment by AviVahl** [comment]
Work left: 2 failing tests in "global scope reference"

**Comment by AviVahl** [comment]
8.0.4 fixed the at-rule exception, but introduced another regression when semi-colons are missing.
I've fixed two tests with the missing semi-colons, and reported the issue upstream.

File: packages/cli/test/build.spec.ts [hunk 1]
```diff
@@ -128,7 +128,7 @@ describe('build stand alone', () => {
         const fs = createFS({
             '/comp.st.css': `
                 :import {
-                    -st-from: "./missing-file.st.css"
+                    -st-from: "./missing-file.st.css";
```
Comment (barak007) [inline packages/cli/test/build.spec.ts:131]: nice catch

File: packages/core/test/stylable-resolver.spec.ts [hunk 2]
```diff
@@ -465,7 +465,7 @@ describe('stylable-resolver', () => {
                     namespace: 'A',
                     content: `
                         :import {
-                            -st-from: "b/index.st.css"
+                            -st-from: "b/index.st.css";
```
Comment (barak007) [inline packages/core/test/stylable-resolver.spec.ts:468]: nice catch

**Review by barak007 — COMMENTED** [review]
10x Avi. This PR looks fine but I saw that the new version dose not allow to extend ast nodes. I don't know that this mean exactly but we are adding keys to the ast nodes. It might be still ok but I need to check it before merging.

**Review by tomrav — APPROVED** [review]
Really nice upgrade

**Comment by AviVahl** [comment]
Approved by @barak007 as well.
Rebasing and will merge once tests pass

**Pull request merged**